orderer/consensus/smartbft/chain.go

```diff
@@ -321,6 +326,37 @@ func (c *BFTChain) Deliver(proposal types.Proposal, signatures []types.Signature
 	c.support.WriteBlock(block, nil)
 }
 
+func (c *BFTChain) reportIsLeader(proposal *types.Proposal) {
+	var nodes []uint64
+	for _, n := range c.RemoteNodes {
+		nodes = append(nodes, n.ID)
+	}
+	nodes = append(nodes, c.Config.SelfID)
+	sort.Slice(nodes, func(i, j int) bool {
+		return nodes[i] < nodes[j]
+	})
+	n := uint64(len(nodes))
+
+	var viewNum uint64
+	if proposal.Metadata == nil { // genesis block
+		viewNum = 0
+	} else {
+		proposalMD := &smartbftprotos.ViewMetadata{}
+		if err := proto.Unmarshal(proposal.Metadata, proposalMD); err != nil {
+			c.Logger.Panicf("Failed unmarshaling smartbft metadata from proposal: %v", err)
+		}
+		viewNum = proposalMD.ViewId
+	}
+
+	leaderID := nodes[viewNum%n] // same calculation as done in the library
+
+	if leaderID == c.Config.SelfID {
+		c.Metrics.IsLeader.Set(1)
```

Comment:
I think it's best just to send the index of the leader.

Reply:
Actually, let's have both values : am I the leader, like you did, and also the index of the leader ?